### PR TITLE
Set WASM_BUILD_WORKSPACE_HINT before starting CI jobs

### PR DESCRIPTION
**Pull request by joao-paulo-parity**
# Problem

CI jobs currently display warnings such as those ones:

> warning: Could not find `Cargo.lock` for `/builds/parity/mirrors/substrate/bin/node-template/runtime/Cargo.toml`, while searching from `/cargo_target_dir/debug/build/node-template-runtime-f76d399030dc21d7/out`. To fix this, point the `WASM_BUILD_WORKSPACE_HINT` env variable to the directory of the workspace being compiled.

> warning: Could not find `Cargo.lock` for `/builds/parity/mirrors/substrate/bin/node/runtime/Cargo.toml`, while searching from `/cargo_target_dir/debug/build/kitchensink-runtime-ebd00ce29afae0e3/out`. To fix this, point the `WASM_BUILD_WORKSPACE_HINT` env variable to the directory of the workspace being compiled.

Despite those warnings, the jobs still succeed.

# Solution

Set up `WASM_BUILD_WORKSPACE_HINT="$PWD"` before starting the jobs.

### CODE REVIEW

**Comment by joao-paulo-parity** [comment]
By https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2105754#L1490 I think we need an adjustment for https://github.com/paritytech/pipeline-scripts/blob/1978a96e208a06866396492394b85940ae6a5444/check_dependent_project.sh#L436. Thankfully it's a simple one.

**Comment by joao-paulo-parity** [comment]
Moving to Draft to prevent accidental merges until https://github.com/paritytech/pipeline-scripts/pull/87 is merged.

**Review by bkchr — APPROVED** [review]

**Comment by joao-paulo-parity** [comment]
Moving to Draft again because I messed up the variable name in https://github.com/paritytech/pipeline-scripts/pull/87. My bad.

**Review by sergejparity — APPROVED** [review]

File: scripts/ci/gitlab/pipeline/publish.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@
     IMAGE_NAME:                    docker.io/$IMAGE_PATH
   before_script:
     - !reference [.job-switcher, before_script]
```
Comment (alvicsam) [inline scripts/ci/gitlab/pipeline/publish.yml:15]:
```suggestion
```
This on can be removed because it's already in `[.kubernetes-env, before_script]`

**Review by alvicsam — CHANGES_REQUESTED** [review]

**Review by alvicsam — APPROVED** [review]

**Comment by joao-paulo-parity** [comment]
bot merge

**Pull request merged**